File: 682/baseball-game.py
class Solution:
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        valid = []
        score = 0
        for op in ops:
            if op == "C":
                score -= valid.pop()
            elif op == "D":
                valid.append(valid[len(valid)-1] * 2)
                score += valid[-1]
            elif op == "+":
                valid.append(valid[-1] + valid[-2])
                score += valid[-1]
            else:
                valid.append(int(op))
                score += valid[-1]
        return score

File: 682/baseball-game.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function (ops) {
    let valid = [];
    let score = 0;
    for (let op of ops) {
        if (op == "C") {
            score -= valid.pop();
        } else if (op == "D") {
            valid.push(valid[valid.length - 1] * 2);
            score += valid[valid.length - 1];
        } else if (op == "+") {
            valid.push(valid[valid.length - 1] + valid[valid.length - 2]);
            score += valid[valid.length - 1];
        } else {
            valid.push(parseInt(op));
            score += valid[valid.length - 1];
        }
    }
    return score;
};

function isNumber(str) {
    return /^\d+$/.test(str);
}